From d9f68455c6533b082a0aa42466dc425e82b6b8c1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20S=C5=82awi=C5=84ski?=
 <amadeuszx.slawinski@intel.com>
Date: Thu, 3 Jan 2019 15:51:30 +0100
Subject: [PATCH 243/292] ASoC: Topology: Fix memory leak from
 soc_tplg_dapm_widget_create
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In soc_tplg_dapm_widget_create() we do unnecessary kstrdup,
we can just directly point to already existing string as it is only used
as template and it's being kstrdup for actual use in
snd_soc_dapm_new_control_unlocked anyway.

Change-Id: I3f610877edb3b636ec8a38a820ec857cc790aad7
Signed-off-by: Amadeusz Sławiński <amadeuszx.slawinski@intel.com>
Reviewed-on:
Reviewed-by: Lewandowski, Gustaw <gustaw.lewandowski@intel.com>
---
 sound/soc/soc-topology.c | 14 ++------------
 1 file changed, 2 insertions(+), 12 deletions(-)

diff --git a/sound/soc/soc-topology.c b/sound/soc/soc-topology.c
index a76423fcfcb0..652efee3b0df 100644
--- a/sound/soc/soc-topology.c
+++ b/sound/soc/soc-topology.c
@@ -1486,15 +1486,8 @@ static int soc_tplg_dapm_widget_create(struct soc_tplg *tplg,
 	if (template.id < 0)
 		return template.id;
 
-	/* strings are allocated here, but used and freed by the widget */
-	template.name = kstrdup(w->name, GFP_KERNEL);
-	if (!template.name)
-		return -ENOMEM;
-	template.sname = kstrdup(w->sname, GFP_KERNEL);
-	if (!template.sname) {
-		ret = -ENOMEM;
-		goto err;
-	}
+	template.name = w->name;
+	template.sname = w->sname;
 	template.reg = w->reg;
 	template.shift = w->shift;
 	template.mask = w->mask;
@@ -1600,9 +1593,6 @@ static int soc_tplg_dapm_widget_create(struct soc_tplg *tplg,
 	snd_soc_tplg_widget_remove(widget);
 	snd_soc_dapm_free_widget(widget);
 hdr_err:
-	kfree(template.sname);
-err:
-	kfree(template.name);
 	return ret;
 }
 
-- 
2.21.0

